mainhandler/imageregistryhandler.go

```diff
@@ -529,3 +595,159 @@ func (registryScan *registryScan) createTriggerRequestCronJob(k8sAPI *k8sinterfa
 	}
 	return nil
 }
+
+func (registryScan *registryScan) parseRegistryAuthFromSecret() error {
+	secret, err := registryScan.getRegistryScanSecret()
+	if err != nil {
+		return err
+	}
+	secretData := secret.GetData()
+	var registriesAuth []registryAuth
+	registriesAuthStr, ok := secretData[registriesAuthFieldInSecret].(string)
+	if !ok {
+		return fmt.Errorf("error parsing Secret: %s field must be a string", registriesAuthFieldInSecret)
+	}
+	data, err := base64.StdEncoding.DecodeString(registriesAuthStr)
+	if err != nil {
+		return fmt.Errorf("error parsing Secret: %s", err.Error())
+	}
+	registriesAuthStr = strings.Replace(string(data), "\n", "", -1)
+	err = json.Unmarshal([]byte(registriesAuthStr), &registriesAuth)
+	if err != nil {
+		return fmt.Errorf("error parsing Secret: %s", err.Error())
+	}
+	//try to find an auth with the same registry name from the request
+	for _, auth := range registriesAuth {
+		if auth.Registry == registryScan.registryInfo.RegistryName {
+			if err := auth.initDefaultValues(); err != nil {
+				return err
+			}
+			registryScan.setRegistryInfoFromAuth(auth, &registryScan.registryInfo)
+			return nil
+		}
+	}
+	//couldn't find auth with the full, check if there is an auth for the registry without the project name
+	regAndProject := strings.Split(registryScan.registryInfo.RegistryName, "/")
+	if len(regAndProject) > 1 {
+		for _, auth := range registriesAuth {
+			if auth.Registry == regAndProject[0] {
+				if err := auth.initDefaultValues(); err != nil {
+					return err
+				}
+				registryScan.setRegistryInfoFromAuth(auth, &registryScan.registryInfo)
+				return nil
+			}
+		}
+
+	}
+	//no auth found for registry return a default one
+	auth := makeRegistryAuth(registryScan.registryInfo.RegistryName)
+	registryScan.setRegistryInfoFromAuth(auth, &registryScan.registryInfo)
+	return nil
+}
+
+func (registryScan *registryScan) setRegistryInfoFromAuth(auth registryAuth, registryInfo *armotypes.RegistryInfo) {
+	registryInfo.AuthMethod.Type = auth.AuthMethod
+	registryInfo.AuthMethod.Username = auth.Username
+	registryInfo.AuthMethod.Password = auth.Password
+	registryInfo.SkipTLSVerify = *auth.SkipTLSVerify
+	registryInfo.IsHTTPs = !*auth.Insecure
+	registryInfo.Kind = string(auth.Kind)
+}
+
+func (registryScan *registryScan) getRegistryConfig(registryInfo *armotypes.RegistryInfo) (string, error) {
```

Comment:
Is this code will give a precedence to the data from notification uppon the data from CM/secret

Reply:
?

Reply:
This flow will only be called if we checked the command and it is the old one, meaning we need to take all info from cluster